measfunc/dmmqdac.py

```diff
@@ -2,182 +2,432 @@
 from qcodes.instrument.base import Instrument
 from qcodes.instrument.channel import InstrumentChannel
 from qcodes import Parameter, ParameterWithSetpoints
-from qcodes.utils.validators import Numbers, Arrays
+from qcodes.utils.validators import Numbers, Ints, Arrays
 from typing import Any, Iterable, Tuple, Union, Optional
 from time import sleep
 
+class BufferedAcquisitionController(Instrument):
+    """
+    Meant to be used with a QDac I and a Keysight digital multimeter. 
+
+    Usage:
+        # # 
+        # # Pre-init config 
+        dSETlb_dbsT = 0.0225 
+        fast_channel_setpoints = {'qdac_channel': qdac.ch01,
+            'vstart': 1.39,
+            'vend': 1.81,
+            'num_samples': 201}
+        fast_compensating_channel_setpoints = {'qdac_channel': qdac.ch02,
+            'vstart': 1.0 + dSETlb_dbsT*(fast_channel_setpoints['vend'] - fast_channel_setpoints['vstart'])/2,
+            'vend': 1.0 - dSETlb_dbsT*(fast_channel_setpoints['vend'] - fast_channel_setpoints['vstart'])/2,
+            'num_samples': 201}
+        # # 
+        # # Initialize class object 
+        trace_acquisition = BufferedAcquisitionController(name='trace_acquisition', 
+            dmm=rf_X, qdac=qdac_A, 
+            fast_channel_setpoints=fast_channel_setpoints,
+            fast_compensating_channel_setpoints=fast_compensating_channel_setpoints)
+    
+        # # 
+        # # Acquire data 
+        ydata = trace_acquisition.buffered_1d_acquisition()
```

Comment:
I think  you should acquire data by using the do0d(trace_acquisition.buffered_1d_acquisition, do_plot = True)